CMakeLists.txt

```diff
@@ -0,0 +1,152 @@
+cmake_minimum_required(VERSION 2.8.12)
+
+project(cctz)
+
+set(CMAKE_MODULE_PATH
+  ${PROJECT_SOURCE_DIR}/cmake
+  ${PROJECT_SOURCE_DIR}/cmake/modules
+  ${CMAKE_MODULE_PATH})
+
+include(CheckCXXCompilerFlag)
+include(CTest)
+include(GNUInstallDirs)
+
+option(BUILD_EXAMPLES "Whether or not to build examples" ON)
+
+if (CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
+  add_compile_options(-Wall)
+endif()
+
+# if a specific standard is required, this can be set from the command line:
+#     cmake -DCMAKE_CXX_STANDARD=[11|14|17]
+if (NOT CMAKE_CXX_STANDARD)
+  # C++11 is the minimum version required
+  set(CMAKE_CXX_STANDARD 11)
```

Comment:
Would it be better to set the property on the target (also supported as of v3.1, which is the only time this takes effect anyway) so that it is transitive? I'm assuming the public headers also require C++11.

Reply:
Ah! Thanks for making me look into this.

It is my understanding that CMAKE_CXX_STANDARD is more targeted to toolchain files than for the projects directly<sup>[1](#footnote-1)</sup>, so this would be an improvement.

I thought the way to do this per-target was:

```cmake
target_compile_features(cctz PUBLIC cxx_std_11)
```

But this is only available since CMake 3.8.
However, you right, because an alternative syntax exists:

```cmake
set_property(TARGET cctz PROPERTY CXX_STANDARD 11)
```

Will make the change once I find something that can work ok with CMake < 3.1,
and if possible respects CMAKE_CXX_STANDARD.

---

<a name="footnote-1">1</a>: https://gitlab.kitware.com/cmake/cmake/issues/17146#note_299405